src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -0,0 +1,46 @@
+
+using System;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// The implementation of the "New-TemporaryFile" cmdlet
+    /// </summary>    
+    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
+    [OutputType(typeof(System.IO.FileInfo))]
+    public class NewTemporaryFileCommand : Cmdlet
+    {      
+        /// <summary>
+        /// returns a TemporaryFile
+        /// </summary>
+        protected override void BeginProcessing()
```

Comment:
Yes I will move it.

I left as is because this is an existing cmdlet and did not want to potentially alter any existing implementations (and I did not feel that I could fully comprehend what those possible scenarios would have been).